pyface/data_view/i_data_view_widget.py

```diff
@@ -8,52 +8,283 @@
+        raise NotImplementedError()
+
+    def _set_control_selection_mode(self, selection_mode):
+        """ Toolkit specific method to change the selection mode. """
+        raise NotImplementedError()
+
+    def _selection_updated(self, event):
+        """ Observer for selection trait. """
+        if self.control is not None and not self._selection_updating_flag:
+            with self._selection_updating():
+                self._set_control_selection(self.selection)
+
+    def _get_control_selection(self):
+        """ Toolkit specific method to get the selection. """
+        raise NotImplementedError()
+
+    def _set_control_selection(self, selection):
+        """ Toolkit specific method to change the selection. """
+        raise NotImplementedError()
+
+    def _observe_control_selection(self, remove=False):
+        """ Toolkit specific method to watch for changes in the selection. """
+        raise NotImplementedError()
+
+    def _update_selection(self, *args, **kwargs):
+        if not self._selection_updating_flag:
+            with self._selection_updating():
+                self._selection = self._get_control_selection()
+
+    # ------------------------------------------------------------------------
+    # Widget Interface
+    # ------------------------------------------------------------------------
+
+    def _create(self):
+        """ Creates the toolkit specific control.
+
+        This method should create the control and assign it to the
+        :py:attr:``control`` trait.
+        """
+        self.control = self._create_control(self.parent)
+        self._initialize_control()
+        self._add_event_listeners()
+
+        self.show(self.visible)
+        self.enable(self.enabled)
+
+    def _initialize_control(self):
+        """ Initializes the toolkit specific control.
+        """
+        logger.debug('Initializing DataViewWidget')
+        self._set_control_header_visible(self.header_visible)
+        self._set_control_selection_mode(self.selection_mode)
+        self._set_control_selection_type(self.selection_type)
+
     def _add_event_listeners(self):
+        logger.debug('Adding DataViewWidget listeners')
         super()._add_event_listeners()
         self.observe(
-            self._header_visible_updated, 'header_visible', dispatch='ui')
+            self._header_visible_updated,
+            'header_visible',
+            dispatch='ui',
+        )
+        self.observe(
+            self._selection_type_updated,
+            'selection_type',
+            dispatch='ui',
+        )
+        self.observe(
+            self._selection_mode_updated,
+            'selection_mode',
+            dispatch='ui',
+        )
+        self.observe(
+            self._selection_updated,
+            '_selection.items',
+            dispatch='ui',
+        )
+        if self.control is not None:
+            self._observe_control_selection()
```

Comment:
It could, but with the current implementation if we were to write a new backend and forget to implement this, they get an error, if the forget to extend `_add/_remove_event_listeners` then there is no warning that a required things has been forgotten.